src/EcommerceStatsBundle/Service/TopSellerService.php

```diff
@@ -0,0 +1,84 @@
+<?php declare(strict_types=1);
+
+namespace ChameleonSystem\EcommerceStatsBundle\Service;
+
+use ChameleonSystem\EcommerceStatsBundle\DataModel\ShopOrderItemDataModel;
+use ChameleonSystem\EcommerceStatsBundle\Interfaces\TopSellerServiceInterface;
+use Doctrine\DBAL\Connection;
+
+class TopSellerService implements TopSellerServiceInterface
+{
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    /**
+     * @return ShopOrderItemDataModel[]
+     */
+    public function getTopsellers(
+        ?string $startDate,
```

Comment:
Could be a DateTime

(and wouldn't need the "23:59:59" below then)

Reply:
:heavy_check_mark: In the meantime I did change it to be a DateTime.

Reply:
You then allow arbitrary times (and not 0:00 to 23:59).

Reply:
True, allowing arbitrary times in the service and ensuring that they are `00:00` and `23:59` in the controller / module seems like a good idea in that case.

Reply:
:heavy_check_mark: